Logging Implementation

NxFerrara opened the pull request:
## Changes:
- Added logging to every source file utilizing logging
- Added log messages where relevant

## Considerations:
- Need to have `.app/logs/` folder created during setup
- Although the log messaging has been tested locally, it needs to be tested on actual device to ensure there are no conflicts such as with file permissions
- Needs to be tested/integrated with `logrotate` to ensure the `watchedFileHandler` works as intended (logging should not be corrupted or missing while a rotation occurs)

NxFerrara committed:
First iteration, needs work

I am doing this so I can pull it from my laptop.

kc34251 committed:
Added logrotate script and added logg
ing points within screen and touch sensor

kc34251 committed:
Added WatchedFileHandler and deleted logrotation script

jbesoto committed:
Commit TODO file

kc34251 committed:
Logging Changes

kc34251 committed:
Merge remote-tracking branch 'origin/jbesoto/setup-review' into kc34251/logging

kc34251 committed:
Setup Logging Changes, moving on from class-based

kc34251 committed:
Test_logging file

NxFerrara committed:
Add logging configuration file

NxFerrara committed:
Remove src/logging

This is no longer needed since we are setting up logging in each module directly.

NxFerrara committed:
Use logging config in test_logging.py

Instead of directly setting up the configurations in the python script, we are testing using the configuration file.

NxFerrara committed:
Add project directory to paths.py

NxFerrara committed:
Add logging to all source files

Set up each module with its own logger and add all relevant logging messages.

NxFerrara committed:
Merge branch 'feature/logging' into nxferrara/logging

kc34251 commented on src/cloud_db/cloud_db.py:
```diff
@@ -60,13 +63,15 @@ def insert_device_data(self, device_id: str) -> APIResponse:
             'name': ''
         }
         response = self.client.table('devices').insert(device_record).execute()
+        logger.info(response)  # TODO: Correctly print response (need to test)
```
Consider adding a string before the response variable to show that this is a supabase api response. EX: "Response: "

kc34251 commented on src/cloud_db/cloud_db.py:
```diff
@@ -40,12 +42,13 @@ def insert_tap_data(self, tap: Tap) -> APIResponse:
             'device_id': tap.device_id
         }
         response = self.client.table('tap_data').insert(tap_record).execute()
+        logger.info(response)  # TODO: Correctly print response (need to test)
```
Same as below

kc34251 commented on src/cloud_db/cloud_db.py:
```diff
@@ -60,13 +63,15 @@ def insert_device_data(self, device_id: str) -> APIResponse:
         Returns:
             An APIResponse object representing the result of the database operation.
         """
+        logger.info('Insertting device record to Supabase')  # TODO: Correctly print response (need to test)
```
Spelling

kc34251 commented on src/screen/screen.py:
Decide whether or not to put individual logging messages inside each display-related function for the purposes of debugging. This would cause a lot of extra log messages but could be useful if there are errors in screen processes. @jbesoto

kc34251 commented on src/touch_sensor/touch_sensor.py:
```diff
@@ -88,28 +85,27 @@ def pipe_tap_data(self, tap: Tap) -> None:
-            logger.error("Named pipe not found")
-            raise FileNotFoundError("Named pipe not found")
+            logger.error('Named pipe not found')
+            raise FileNotFoundError('Named pipe not found')
```
Why is an error raised for FileNotFound but not for any other errors. @jbesoto

kc34251 commented on src/utils/utils.py:
```diff
@@ -1,10 +1,16 @@
 import fcntl
 import subprocess
+import logging
+import os
```
No logging.config import

kc34251 commented on src/utils/utils.py:
```diff
@@ -45,5 +51,4 @@ def get_device_id() -> str:
-        print(f"An error occurred while fetching the hostname: {e}")
-        raise
\ No newline at end of file
+        logger.error(f"An error occurred while fetching the hostname: {e}")
```
why is the raise removed.

jbesoto committed:
Bugfix: Create .app folder

jbesoto committed:
Bugfix: Add jq to the list of dependencies

'jq' is used by 3_set_device_id.sh for parsing the JSON received from
the Supabase API and extracting the first device ID. 'jq' does not come
pre-installed with the Libretech ISO image.

jbesoto committed:
Bugfix: Typo fixed on main

';' was missing on exit command. This caused a syntax error due to
misinterpreted brackets.

jbesoto committed:
Bugfix: Refactoring reboot behavior

Executing 'reboot' command would not reboot the system immediately. As a
result, `setup.sh` would execute an additional iteration of the
reboot-prompt loop and print the prompt again. This was fixed by adding
a break after reboot.

NxFerrara committed:
Merge branch 'jbesoto/setup-review' into nxferrara/logging

jbesoto committed:
Move 3_install_dependencies to pre-reboot stage

Some dependencies, like 'jq', are required by some pre-reboot scripts.

NxFerrara committed:
Merge branch 'jbesoto/setup-review' into nxferrara/logging

jbesoto committed:
Add function for creating app directories

NxFerrara committed:
Merge branch 'jbesoto/setup-review' into nxferrara/logging

jbesoto committed:
Move 3_create_venv script to pre-reboot phase

NxFerrara committed:
Merge branch 'jbesoto/setup-review' into nxferrara/logging

jbesoto committed:
Bugfix: Call function for installing Unix packages

It was not called anywhere, so 'jq' was never installed.

jbesoto committed:
Remove 2_set_user_permissions.sh

It was decided that creating a user and a group was unnecessary as root
permissions are required to run the system. In this case, creating a
user and setting specific permissions is redundant. So, we decided to
change the user on which the system services are run to root.

NxFerrara committed:
Merge branch 'jbesoto/setup-review' into nxferrara/logging

jbesoto committed:
Modify prompting behavior for WiFi credentials

With this new implementation, the user is prompted first whether they
want to connect to WiFi. If yes, they are then prompted for the WiFi
credentials.

NxFerrara committed:
Merge branch 'jbesoto/setup-review' into nxferrara/logging

jbesoto committed:
Bugfix: Fix syntax error in 'if' block

NxFerrara committed:
Merge branch 'jbesoto/setup-review' into nxferrara/logging

jbesoto committed:
Fixbug: Close if statement

NxFerrara committed:
Merge branch 'jbesoto/setup-review' into nxferrara/logging
